### ISSUE TITLE
[🐛 Bug]: essage: Could not start a new session. Could not start a new session. Error while creating session with the driver service

### ISSUE THREAD

**Issue by jakobdo** [SeleniumHQ/selenium#13506]
### What happened?

I am trying to start a new session in a selenium grid and from time to time, I am getting this error:

selenium.common.exceptions.SessionNotCreatedException: Message: Could not start a new session. Could not start a new session. Error while creating session with the driver service. Stopping driver service: Could not start a new session. Response code 500. Message: Failed to decode response from marionette  
Host info: host: 'SRVP01234', ip: '192.1168.1.68'
Build info: version: '4.16.1', revision: '9b4c83354e'
System info: os.name: 'Windows Server 2022', os.arch: 'amd64', os.version: '10.0', java.version: '21.0.1'
Driver info: driver.version: unknown
Build info: version: '4.16.1', revision: '9b4c83354e'
System info: os.name: 'Windows Server 2022', os.arch: 'amd64', os.version: '10.0', java.version: '21.0.1'
Driver info: driver.version: unknown
Build info: version: '4.16.1', revision: '9b4c83354e'
System info: os.name: 'Windows Server 2022', os.arch: 'amd64', os.version: '10.0', java.version: '21.0.1'
Driver info: driver.version: unknown
Stacktrace:
    at org.openqa.selenium.grid.node.remote.RemoteNode.newSession (RemoteNode.java:151)
    at org.openqa.selenium.grid.distributor.local.LocalDistributor.startSession (LocalDistributor.java:645)
    at org.openqa.selenium.grid.distributor.local.LocalDistributor.newSession (LocalDistributor.java:564)
    at org.openqa.selenium.grid.distributor.local.LocalDistributor$NewSessionRunnable.handleNewSessionRequest (LocalDistributor.java:824)
    at org.openqa.selenium.grid.distributor.local.LocalDistributor$NewSessionRunnable.lambda$run$1 (LocalDistributor.java:784)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)

### How can we reproduce the issue?

```shell
I am trying to start a session using python and this simplified code:


from selenium import webdriver
from selenium.webdriver import FirefoxOptions

with webdriver.Remote(
        command_executor="http://192.168.1.60:4444",
        options=FirefoxOptions(),
        file_detector=file_detector
    ) as driver:
    driver.get("http://google.com")

Config for nodes are:
[node]
driver-implementation = ["chrome", "firefox"]
hub = "selenium-grid.domain.local"
selenium-manager = true
max-sessions = 5
override-max-sessions = true
session-timeout = 60

I have tried with default settings for max-sessions and override-max-sessions, which gave same result.
```

### Relevant log output

```shell
selenium.common.exceptions.SessionNotCreatedException: Message: Could not start a new session. Could not start a new session. Error while creating session with the driver service. Stopping driver service: Could not start a new session. Response code 500. Message: Failed to decode response from marionette  
Host info: host: 'SRVP01234', ip: '192.1168.1.68'
Build info: version: '4.16.1', revision: '9b4c83354e'
System info: os.name: 'Windows Server 2022', os.arch: 'amd64', os.version: '10.0', java.version: '21.0.1'
Driver info: driver.version: unknown
Build info: version: '4.16.1', revision: '9b4c83354e'
System info: os.name: 'Windows Server 2022', os.arch: 'amd64', os.version: '10.0', java.version: '21.0.1'
Driver info: driver.version: unknown
Build info: version: '4.16.1', revision: '9b4c83354e'
System info: os.name: 'Windows Server 2022', os.arch: 'amd64', os.version: '10.0', java.version: '21.0.1'
Driver info: driver.version: unknown
Stacktrace:
    at org.openqa.selenium.grid.node.remote.RemoteNode.newSession (RemoteNode.java:151)
    at org.openqa.selenium.grid.distributor.local.LocalDistributor.startSession (LocalDistributor.java:645)
    at org.openqa.selenium.grid.distributor.local.LocalDistributor.newSession (LocalDistributor.java:564)
    at org.openqa.selenium.grid.distributor.local.LocalDistributor$NewSessionRunnable.handleNewSessionRequest (LocalDistributor.java:824)
    at org.openqa.selenium.grid.distributor.local.LocalDistributor$NewSessionRunnable.lambda$run$1 (LocalDistributor.java:784)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
```

### Operating System

Windows Server 2022

### Selenium version

Python 4.17.1

### What are the browser(s) and version(s) where you see this issue?

Firefox 122 64bit, Chrome 121.0.6167.86 64bit

### What are the browser driver(s) and version(s) where you see this issue?

Firefox GeckoDriver 0.34 (I am using this option: selenium-manager = true)

### Are you using Selenium Grid?

Yes: 4.16.1

**Comment by diemol** [comment]
Can you share the complete log from the Grid? Also the command you use to start the Grid.

**Comment by jakobdo** [comment]
Hello diemol, and thanks for taking the time to look into this issue.
But I have to admit, I was able to re-create the issue the following way:
Start session 1. (kill it before calling driver.close() / driver.quit() )
Start session 2 and I got the above error.
But after I have enabled logging, I am not able to get the error at all.
So I have to say NO.

The selenium grid is running as a service, but started with this command:

java -jar selenium-server.jar node --config config.toml
selenium-server.jar = 4.16.1
config.toml:
[node]
driver-implementation = ["chrome", "firefox"]
hub = "selenium-grid.domain.local"
selenium-manager = true
session-timeout = 60

[logging]
log-file = 'C:\selenium\logs\gridnode.log'

And I have tried different configs for more sessions to force an error, but not this time.

**Comment by jakobdo** [comment]
I am just informed, that the servers are running using dynamic memory, so the problem could be related to the servers having to small amount of ram/memory for a start, but now they are boosted and supports multiple sessions.

**Comment by jakobdo** [comment]
We have now changed the servers memory to fixed/static memory. 6GB pr server.
And I have been able to get the error while logging was enabled. How much of the log do you want?
I do not want to share too much details about the server/setup. (from a security point of view)

**Comment by jakobdo** [comment]
Please find attached log from node.

[github.selenium.grid.log](https://github.com/SeleniumHQ/selenium/files/14063393/github.selenium.grid.log)

**Comment by jakobdo** [comment]
Do you guys need more input/information from my side, to investigate this issue?

**Comment by diemol** [comment]
I can see this in the logs:
>Response code 500. Message: Failed to decode response from marionette 

Which points to an error in GeckoDriver. After Googling, a workaround says that increasing memory helps (in case the host has limited resources).

I am not sure we can give any more pointers because you already mentioned above that the situation improved when more memory was allocated. 

You should collect some CPU and memory metrics from your infrastructure and correlate when the issue happens and what resources were being used.

**Comment by jakobdo** [comment]
Hello again, thanks for taking your time once again to look into this issue.
I have now tried to bump the memory to 16GB on one of the nodes and stopped the service on the others.
I am getting some errors in the startup, I am unsure if this is unrelated.
(see attached file: [grid.log](https://github.com/SeleniumHQ/selenium/files/14097239/grid.log))

I tried to contact the selenium grid and once again I got the same error, that started this issue. Then I restarted the service and it worked.
I will now try to restart the server and see if I can get the error once more. 
When I am getting the error, the memory usage is close to 3-4gb, and not going up, when I am getting the error.
